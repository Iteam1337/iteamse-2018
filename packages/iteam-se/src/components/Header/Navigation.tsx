import * as React from 'react'
import { withApollo, WithApolloClient } from 'react-apollo'
import { Link, NavLink } from 'react-router-dom'
import SVG from 'react-svg'
import { ABOUT_PAGE_QUERY } from '../../pages/About'
import { CASES_PAGE_QUERY } from '../../pages/Cases'
import { HOME_PAGE_QUERY } from '../../pages/Home'
import { HOW_WE_WORK_PAGE_QUERY } from '../../pages/HowWeWork'
import { OFFERS_PAGE_QUERY } from '../../pages/Offers'
import { TEAM_PAGE_QUERY } from '../../pages/Team'
import { WORK_PAGE_QUERY } from '../../pages/Work'
import styled, { createGlobalStyle, keyframes } from '../../theme'
import NativeLink from '../Link/NativeLink'
import logo from './img/iteam.svg'
import logoBlack from './img/iteam_black.svg'

interface NavigationProps {
  isInverted?: boolean
  noShadow?: boolean
}
interface NavigationState {
  indicatorLocation: number
  indicatorWidth: number
}

const Wrap = styled.div`
  display: grid;
  grid-column: 2;
  grid-row: 1;
  grid-template-columns: 150px 1fr;

  @media (min-width: 481px) {
    padding: 0;
  }

  ${({ theme }) =>
    theme.browsers.ie10Or11(`
    display: flex;
    justify-content: space-between;
  `)};
`

const LogoLink = styled(Link)`
  &:focus {
    outline: none;
  }
`

const Logo = styled(SVG)`
  > * svg {
    max-width: 75%;
  }

  @media (min-width: 1025px) {
    > * svg {
      max-width: 100%;
    }
  }
`

const slideDown = keyframes`
  from {
    opacity: 0;
    transform: translateY(-5px);
  }

  to {
    opacity: 1;
    transform: translateY(0px);
  }
`

interface IndicatorProps {
  indicatorLocation: number
  indicatorWidth: number
}

const Indicator =
  styled.div <
  IndicatorProps >
  `
  animation: ${slideDown} 300ms ease-in-out 1;
  background-color: #fff;
  top: 0;
  height: 5px;
  left: ${({ indicatorLocation }) => `${indicatorLocation}px`};
  position: absolute;
  width: ${({ indicatorWidth }) => `${indicatorWidth}px`};
`

const StyledLink = styled(NavLink)`
  font-size: 18px;
  font-weight: 400;
  text-decoration: none;
  transition: color 200ms ease-in-out;

  &:not(:last-child) {
    margin-right: 30px;
  }

  &:focus {
    outline: none;
  }
`

const ExternalLink = styled(NativeLink)`
  font-size: 18px;
  font-weight: 400;
  text-decoration: none;
  transition: color 200ms ease-in-out;

  &:not(:last-child) {
    margin-right: 30px;
  }

  &:focus {
    outline: none;
  }
`

const NavigationItems =
  styled.div <
  NavigationProps >
  `
  display: none;
  justify-self: flex-end;
  padding-bottom: 40px;
  padding-top: 40px;
  position: relative;

  @media (min-width: 1025px) {
    display: block;
  }

  ${StyledLink}, ${ExternalLink} {
    color: ${({ isInverted }) => (isInverted ? '#000' : '#fff')};
    text-shadow: ${({ noShadow }) =>
      noShadow ? 'none' : `0px 1px 14px rgba(0, 0, 0, 0.5)`};

    &:hover {
      color: ${({ isInverted }) => (isInverted ? '#444' : '#ccc')};
    }
  }
`

const GlobalStyle = createGlobalStyle`
  html[data-whatinput="keyboard"] {
    ${LogoLink} {
      &:focus {
        outline: 3px solid #54FBAC;
        outline-offset: 3px;
      }
    }

    ${StyledLink} {
      &:focus {
        border-bottom: 3px solid #54FBAC;
        color: #54FBAC;
      }
    }
  }
`

export class Navigation extends React.Component<
  WithApolloClient<NavigationProps>,
  NavigationState
> {
  state = {
    indicatorLocation: 0,
    indicatorWidth: 0,
  }

  componentDidMount() {
    const active = document.querySelector('.active-nav') as HTMLAnchorElement

    if (active) {
      this.setState({
        indicatorLocation: active.offsetLeft,
        indicatorWidth: active.offsetWidth,
      })
    }
  }

  prefetchPage = (page: string) => () => {
    let query

    switch (page) {
      case 'home':
        query = HOME_PAGE_QUERY
        break
      case 'about':
        query = ABOUT_PAGE_QUERY
        break
      case 'case':
        query = CASES_PAGE_QUERY
        break
      case 'how-we-work':
        query = HOW_WE_WORK_PAGE_QUERY
        break
      case 'team':
        query = TEAM_PAGE_QUERY
        break
      case 'work':
        query = WORK_PAGE_QUERY
        break
      case 'erbjudanden':
        query = OFFERS_PAGE_QUERY
        break
      default:
        break
    }

    if (query) {
      this.props.client.query({
        query,
      })
    }
  }

  render() {
    const { isInverted = false, noShadow = false } = this.props
    const { indicatorLocation, indicatorWidth } = this.state

    return (
      <Wrap>
        <GlobalStyle />
        <LogoLink onMouseEnter={this.prefetchPage('home')} to="/">
          <Logo data-testid="logo" path={isInverted ? logoBlack : logo} />
        </LogoLink>
        <NavigationItems isInverted={isInverted} noShadow={noShadow}>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('erbjudanden')}
            to="/erbjudanden"
          >
            Erbjudanden
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('case')}
            to="/case"
          >
            Våra case
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('how-we-work')}
            to="/hur-vi-jobbar"
          >
            Metod
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('team')}
            to="/medarbetare"
          >
            Medarbetare
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('work')}
            to="/karriar"
          >
            Karriär
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('about')}
            to="/om-oss"
          >
            Om
          </StyledLink>
          <ExternalLink
            target="_blank"
            rel="noopener noreferrer"
            activeClassName="active-nav"
            href="https://blog.iteam.life/"
          >
            Labs
          </ExternalLink>
          <Indicator
            indicatorLocation={indicatorLocation}
            indicatorWidth={indicatorWidth}
          />
        </NavigationItems>
      </Wrap>
    )
  }
}

export default withApollo<NavigationProps>(Navigation)
