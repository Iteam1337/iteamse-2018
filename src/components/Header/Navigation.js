// @flow

import * as React from 'react'
import styled, { injectGlobal, keyframes } from 'styled-components'
import logo from './img/iteam.svg'
import { NavLink, Link } from 'react-router-dom'
import { withApollo } from 'react-apollo'
import { HomePageQuery } from '../../pages/Home'
import { AboutPageQuery } from '../../pages/About'
import { HowWeWorkPageQuery } from '../../pages/HowWeWork'
import { TeamPageQuery } from '../../pages/Team'
import { OperationsPageQuery } from '../../pages/Ops'
import { WorkPageQuery } from '../../pages/Work'
import { CasesPageQuery } from '../../pages/Cases'
import SVG from 'react-svg'

type Props = {
  client: {
    query: Function,
  },
}

type State = {
  indicatorLocation: number,
  indicatorWidth: number,
}

const Wrap = styled.div`
  align-items: center;
  background-image: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0));
  display: grid;
  grid-template-columns: 150px 1fr;
  padding: 20px;

  @media (min-width: 481px) {
    padding: 0;
  }
`

const LogoLink = styled(Link)`
  &:focus {
    outline: none;
  }
`

const Logo = styled(SVG)`
  max-width: 100%;
`

const NavigationItems = styled.div`
  display: none;
  justify-self: flex-end;
  padding-bottom: 40px;
  padding-top: 40px;
  position: relative;

  @media (min-width: 481px) {
    display: block;
  }
`

const slideDown = keyframes`
  from {
    opacity: 0;
    transform: translateY(-5px);
  }

  to {
    opacity: 1;
    transform: translateY(0px);
  }
`

const Indicator = styled.div`
  animation: ${slideDown} 300ms ease-in-out 1;
  background-color: #fff;
  top: 0;
  height: 5px;
  left: ${({ indicatorLocation }) => `${indicatorLocation}px`};
  position: absolute;
  width: ${({ indicatorWidth }) => `${indicatorWidth}px`};
`

const StyledLink = styled(NavLink)`
  color: #fff;
  font-size: 18px;
  font-weight: 400;
  text-decoration: none;
  text-shadow: 0px 1px 14px rgba(0, 0, 0, 0.5);
  transition: color 200ms ease-in-out;

  &:not(:last-child) {
    margin-right: 30px;
  }

  &:hover {
    color: #ccc;
  }

  &:focus {
    outline: none;
  }
`

injectGlobal`
  html[data-whatinput="keyboard"] {
    ${LogoLink} {
      &:focus {
        outline: 3px solid #54FBAC;
        outline-offset: 3px;
      }
    }

    ${StyledLink} {
      &:focus {
        border-bottom: 3px solid #54FBAC;
        color: #54FBAC;
      }
    }
  }
`

export class Navigation extends React.Component<Props, State> {
  props: Props

  state = {
    indicatorLocation: 0,
    indicatorWidth: 0,
  }

  componentDidMount () {
    const active = document.querySelector('.active-nav')

    if (active) {
      this.setState({
        indicatorLocation: active.offsetLeft,
        indicatorWidth: active.offsetWidth,
      })
    }
  }

  prefetchPage = (page: string) => () => {
    let query

    switch (page) {
      case 'home':
        query = HomePageQuery
        break
      case 'about':
        query = AboutPageQuery
        break
      case 'case':
        query = CasesPageQuery
        break
      case 'how-we-work':
        query = HowWeWorkPageQuery
        break
      case 'team':
        query = TeamPageQuery
        break
      case 'ops':
        query = OperationsPageQuery
        break
      case 'work':
        query = WorkPageQuery
        break
    }

    if (query) {
      this.props.client.query({
        query,
      })
    }
  }

  render () {
    const { indicatorLocation, indicatorWidth } = this.state

    return (
      <Wrap>
        <LogoLink onMouseEnter={this.prefetchPage('home')} to="/">
          <Logo alt="Iteam logo" path={logo} />
        </LogoLink>

        <NavigationItems>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('about')}
            to="/om-oss"
          >
            Om oss
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('how-we-work')}
            to="/hur-vi-jobbar"
          >
            Hur vi jobbar
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('team')}
            to="/teamet"
          >
            Teamet
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('case')}
            to="/case"
          >
            VÃ¥ra case
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('work')}
            to="/jobba-hos-oss"
          >
            Jobba hos oss
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('ops')}
            to="/ops"
          >
            Drift & Support
          </StyledLink>
          <Indicator
            indicatorLocation={indicatorLocation}
            indicatorWidth={indicatorWidth}
          />
        </NavigationItems>
      </Wrap>
    )
  }
}

export default withApollo(Navigation)
