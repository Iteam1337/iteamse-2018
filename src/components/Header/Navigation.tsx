import * as React from 'react'
import { withApollo, WithApolloClient } from 'react-apollo'
import { Link, NavLink } from 'react-router-dom'
import SVG from 'react-svg'
import { ABOUT_PAGE_QUERY } from '../../pages/About'
import { CASES_PAGE_QUERY } from '../../pages/Cases'
import { HOME_PAGE_QUERY } from '../../pages/Home'
import { HOW_WE_WORK_PAGE_QUERY } from '../../pages/HowWeWork'
import { OPERATIONS_PAGE_QUERY } from '../../pages/Ops'
import { TEAM_PAGE_QUERY } from '../../pages/Team'
import { WORK_PAGE_QUERY } from '../../pages/Work'
import styled, { injectGlobal, keyframes, withProps } from '../../theme'
import logo from './img/iteam.svg'

interface NavigationState {
  indicatorLocation: number
  indicatorWidth: number
}

const Wrap = styled.div`
  display: grid;
  grid-column: 2;
  grid-row: 1;
  grid-template-columns: 150px 1fr;

  @media (min-width: 481px) {
    padding: 0;
  }
`

const LogoLink = styled(Link)`
  &:focus {
    outline: none;
  }
`

const Logo = styled(SVG)`
  max-width: 100%;
`

const NavigationItems = styled.div`
  display: none;
  justify-self: flex-end;
  padding-bottom: 40px;
  padding-top: 40px;
  position: relative;

  @media (min-width: 1025px) {
    display: block;
  }
`

const slideDown = keyframes`
  from {
    opacity: 0;
    transform: translateY(-5px);
  }

  to {
    opacity: 1;
    transform: translateY(0px);
  }
`

interface IndicatorProps {
  indicatorLocation: number
  indicatorWidth: number
}

const Indicator = withProps<IndicatorProps>()(styled.div)`
  animation: ${slideDown} 300ms ease-in-out 1;
  background-color: #fff;
  top: 0;
  height: 5px;
  left: ${({ indicatorLocation }) => `${indicatorLocation}px`};
  position: absolute;
  width: ${({ indicatorWidth }) => `${indicatorWidth}px`};
`

const StyledLink = styled(NavLink)`
  color: #fff;
  font-size: 18px;
  font-weight: 400;
  text-decoration: none;
  text-shadow: 0px 1px 14px rgba(0, 0, 0, 0.5);
  transition: color 200ms ease-in-out;

  &:not(:last-child) {
    margin-right: 30px;
  }

  &:hover {
    color: #ccc;
  }

  &:focus {
    outline: none;
  }
`

injectGlobal`
  html[data-whatinput="keyboard"] {
    ${LogoLink} {
      &:focus {
        outline: 3px solid #54FBAC;
        outline-offset: 3px;
      }
    }

    ${StyledLink} {
      &:focus {
        border-bottom: 3px solid #54FBAC;
        color: #54FBAC;
      }
    }
  }
`

export class Navigation extends React.Component<
  WithApolloClient<{}>,
  NavigationState
> {
  state = {
    indicatorLocation: 0,
    indicatorWidth: 0,
  }

  componentDidMount() {
    const active = document.querySelector('.active-nav') as HTMLAnchorElement

    if (active) {
      this.setState({
        indicatorLocation: active.offsetLeft,
        indicatorWidth: active.offsetWidth,
      })
    }
  }

  prefetchPage = (page: string) => () => {
    let query

    switch (page) {
      case 'home':
        query = HOME_PAGE_QUERY
        break
      case 'about':
        query = ABOUT_PAGE_QUERY
        break
      case 'case':
        query = CASES_PAGE_QUERY
        break
      case 'how-we-work':
        query = HOW_WE_WORK_PAGE_QUERY
        break
      case 'team':
        query = TEAM_PAGE_QUERY
        break
      case 'ops':
        query = OPERATIONS_PAGE_QUERY
        break
      case 'work':
        query = WORK_PAGE_QUERY
        break
      default:
        break
    }

    if (query) {
      this.props.client.query({
        query,
      })
    }
  }

  render() {
    const { indicatorLocation, indicatorWidth } = this.state

    return (
      <Wrap>
        <LogoLink onMouseEnter={this.prefetchPage('home')} to="/">
          <Logo data-testid="logo" path={logo} />
        </LogoLink>

        <NavigationItems>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('about')}
            to="/om-oss"
          >
            Om oss
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('how-we-work')}
            to="/hur-vi-jobbar"
          >
            Hur vi jobbar
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('team')}
            to="/teamet"
          >
            Teamet
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('case')}
            to="/case"
          >
            VÃ¥ra case
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('work')}
            to="/jobba-hos-oss"
          >
            Jobba hos oss
          </StyledLink>
          <StyledLink
            activeClassName="active-nav"
            onMouseEnter={this.prefetchPage('ops')}
            to="/ops"
          >
            Drift & Support
          </StyledLink>
          <Indicator
            indicatorLocation={indicatorLocation}
            indicatorWidth={indicatorWidth}
          />
        </NavigationItems>
      </Wrap>
    )
  }
}

export default withApollo<{}>(Navigation)
