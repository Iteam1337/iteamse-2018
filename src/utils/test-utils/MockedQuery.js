import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import ApolloClient from 'apollo-client'
import { createHttpLink } from 'apollo-link-http'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

type P = {
  children: React.Node,
  loading?: boolean,
  response: Object,
}

const MockedQuery = ({ children, loading, response = {} }: P) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    link: createHttpLink({
      uri: 'http://localhost:3000',
      fetch: mockedNetworkFetch,
    }),
    cache: new InMemoryCache(),
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
