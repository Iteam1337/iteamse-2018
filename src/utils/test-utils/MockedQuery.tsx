import { InMemoryCache } from 'apollo-cache-inmemory'
import ApolloClient from 'apollo-client'
import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { mockSingleLink } from 'react-apollo/test-utils'
import MockRouter from 'react-mock-router'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

interface MockedQueryProps {
  location?: object
  mocks: any[]
  params?: object
}

const MockedQuery: React.SFC<MockedQueryProps> = ({
  children,
  location,
  mocks,
  params,
}) => {
  const link = mockSingleLink.apply(null, mocks)

  const client = new ApolloClient({
    cache: new InMemoryCache(),
    link,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <MockRouter location={location} params={params}>
          {children}
        </MockRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
