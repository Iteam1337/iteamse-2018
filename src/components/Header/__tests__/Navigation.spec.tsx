import React from 'react'
import { render, Simulate, wait } from 'react-testing-library'
import MockedQuery from '../../../utils/test-utils/MockedQuery'
import { Navigation } from '../Navigation'

jest.mock('graphql-tag', () => jest.fn(input => input))

describe('components/Navigation', () => {
  let client: any
  const mocks = {}

  beforeEach(() => {
    client = {
      query: jest.fn(),
    }
  })

  it('handle hover of about page', async () => {
    const { getByText } = render(
      <MockedQuery mocks={mocks}>
        <Navigation client={client} />
      </MockedQuery>
    )

    await wait()

    Simulate.mouseEnter(getByText('Om'))

    expect(client.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('handle hover of logo', async () => {
    const { getByTestId } = render(
      <MockedQuery mocks={mocks}>
        <Navigation client={client} />
      </MockedQuery>
    )

    await wait()

    Simulate.mouseEnter(getByTestId('logo').parentNode)

    expect(client.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('handle hover of how we work', async () => {
    const { getByText } = render(
      <MockedQuery mocks={mocks}>
        <Navigation client={client} />
      </MockedQuery>
    )

    await wait()

    Simulate.mouseEnter(getByText('Metod'))

    expect(client.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('handle hover of the team page', async () => {
    const { getByText } = render(
      <MockedQuery mocks={mocks}>
        <Navigation client={client} />
      </MockedQuery>
    )

    await wait()

    Simulate.mouseEnter(getByText('Medarbetare'))

    expect(client.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('handle hover of our projects', async () => {
    const { getByText } = render(
      <MockedQuery mocks={mocks}>
        <Navigation client={client} />
      </MockedQuery>
    )

    await wait()

    Simulate.mouseEnter(getByText('Case'))

    expect(client.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('handle hover of work with us', async () => {
    const { getByText } = render(
      <MockedQuery mocks={mocks}>
        <Navigation client={client} />
      </MockedQuery>
    )

    await wait()

    Simulate.mouseEnter(getByText('KarriÃ¤r'))

    expect(client.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('handle hover of operations', async () => {
    const { getByText } = render(
      <MockedQuery mocks={mocks}>
        <Navigation client={client} />
      </MockedQuery>
    )

    await wait()

    Simulate.mouseEnter(getByText('Drift & Support'))

    expect(client.query.mock.calls[0][0]).toMatchSnapshot()
  })
})
