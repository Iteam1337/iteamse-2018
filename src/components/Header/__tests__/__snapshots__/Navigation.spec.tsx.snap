// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation handle hover of about page 1`] = `
Object {
  "query": Array [
    "
  query AboutPage {
    pageAboutUs {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
      contactTitle
      funText
      funTitle
      goodText
      goodTitle
      imageBleed
      stabilityIcons
      stabilityText
      stabilityTitle
      team
      valueText
      valueTitle
    }
  }
",
  ],
}
`;

exports[`components/Navigation handle hover of how we work 1`] = `
Object {
  "query": Array [
    "
  query HowWeWorkPage {
    openpositions {
      location
      id
      title
    }
    pageHowWeWork {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
      contactTitle
      customersText
      customersTitle
      hiringTitle
      openApplicationText
      openApplicationHeader
      imageBleed
      methodText
      methodTitle
      sharingText
      sharingTitle
      team
      teamText
      teamTitle
    }
  }
",
  ],
}
`;

exports[`components/Navigation handle hover of logo 1`] = `
Object {
  "query": Array [
    "
  query HomePage {
    pageStart {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
      aiTitle
      aiText
      aiSlug
      aiLinkText
      codeMobileImage
      codeText
      codeTitle
      codeLogo
      codeSlug
      codeLinkText
      contactTitle
      cultureMobileImage
      cultureText
      cultureTitle
      cultureLogo
      cultureSlug
      cultureLinkText
      strategyMobileImage
      strategyText
      strategyTitle
      strategyLogo
      strategySlug
      strategyLinkText
      team
    }
  }
",
  ],
}
`;

exports[`components/Navigation handle hover of our projects 1`] = `
Object {
  "query": Array [
    "
  query CasesPage {
    pageCases {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
      contactTitle
      team
    }
    cases {
      location
      shortDescription
      slug
      thumbnailImage
      title
    }
  }
",
  ],
}
`;

exports[`components/Navigation handle hover of the team page 1`] = `
Object {
  "query": Array [
    "
  query TeamPage {
    pageTeam {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
      contactTitle
      team
    }
    team {
      avatar
      email
      location
      name
      phoneNumber
      short
      title
    }
  }
",
  ],
}
`;

exports[`components/Navigation handle hover of work with us 1`] = `
Object {
  "query": Array [
    "
  query WorkPage {
    pageWork {
      headerImage
      headerText1
      headerText2
      headerTextBgColor
      contactTitle
      openApplicationText
      openApplicationLabel
      team
    }
    openpositions {
      id
      location
      role
      title
    }
  }
",
  ],
}
`;
